shared/common-adapters/icon.native.js

```diff
@@ -11,74 +10,47 @@ import {omit} from 'lodash'
 import type {Exact} from '../constants/types/more'
 import type {IconType, Props} from './icon'
 
-class Icon extends Component<void, Exact<Props>, void> {
-  render () {
-    let color = shared.defaultColor(this.props.type)
-    let iconType = shared.typeToIconMapper(this.props.type)
+const Icon = (props: Exact<Props>) => {
+  let iconType = shared.typeToIconMapper(props.type)
 
-    if (!iconType) {
-      console.warn('Null iconType passed')
-      return null
-    }
-
-    color = this.props.style && this.props.style.color || color || (this.props.opacity ? globalColors.lightGrey : globalColors.black_40)
+  if (!iconType) {
+    console.warn('Null iconType passed')
+    return null
+  }
+  if (!iconMeta[iconType]) {
+    console.warn(`Invalid icon type passed in: ${iconType}`)
+    return null
+  }
 
-    const styleWidth = this.props.style && this.props.style.width
+  const color = props.style && props.style.color || shared.defaultColor(props.type) || (props.opacity ? globalColors.lightGrey : globalColors.black_40)
+  const styleWidth = props.style && props.style.width
+  const width = styleWidth && {width: props.style.width}
+  const backgroundColor = props.style && {backgroundColor: props.style.backgroundColor} || {}
 
-    const width = styleWidth && {width: this.props.style.width}
-    const height = this.props.style && this.props.style.height && {height: this.props.style.height}
+  let icon
 
+  if (iconMeta[iconType].isFont) {
     const fontSizeHint = shared.fontSize(iconType)
-    const fontSize = (this.props.style && (this.props.style.fontSize || styleWidth) && {fontSize: this.props.style.fontSize || styleWidth}) || fontSizeHint
-    const textAlign = this.props.style && this.props.style.textAlign
-    const backgroundColor = this.props.style && {backgroundColor: this.props.style.backgroundColor} || {}
-
-    // Extract style props for our container and box
-    let containerProps = omit(this.props.style, [
```

Comment:
lets only extract what we need for our inner stuff and let anything else style the container